docs/src/python/distributed.rst

```diff
@@ -0,0 +1,19 @@
+.. _distributed:
+
+.. currentmodule:: mlx.core.distributed
+
+Distributed Communication
+==========================
+
+MLX provides a distributed communication package using MPI. The MPI library is
+loaded at runtime; if MPI is available then distributed communication is also
+made available.
+
+.. autosummary::
+   :toctree: _autosummary
+
+    Group
+    is_available
+    init
+    all_reduce_sum
```

Comment:
Sorry for the random comment, but the thought just ocurred to me, isn't it more common to just use `all_reduce` as the op name? Presumably the intention with the `sum` suffix is to preserve flexibility (if one wanted a max or prod)?

Reply:
It actually isn't a random comment at all. The common approach is `all_reduce(x, op[, group])`. I kinda went with `all_reduce_<op>(x[, group])` cause it felt a bit better. Kind of in the spirit of having `sum`, `prod`, ... etc instead of `reduce(x, op)`.

Reply:
Perhaps we should remove the reduce... not sure.

Reply:
Yes the thought occurred to me as well, e.g. `all_sum`, `all_max`, ....

In a vacuum I think I prefer that as it's both fewer characters and a bit more consistent with the naming conventinos in core (e.g. `sum`, `max`, ... and not `reduce_sum`, `reduce_max`, ...). But I'm not sure if removing the reduce adds some confusion in this case?

Reply:
Yea you're right, we should keep the `all` in there.

Reply:
So I guess the question is should it be:

`distributed.all_sum` (for all-to-all) and `distributed.sum` for reducing to a single process

or keep it as is and include the `reduce` as a prefix?

Reply:
I am down to change it to `all_sum`, it will also look better as I always use it as `mx.distributed.all_reduce_sum` never as a standalone function.